# Quantization configuration

# quantization types --> uniform discretization, adaptive binning w/ f(x)=x^3, etc.
quant_types: ['uniform','x3','x5']

# number of available bits
available_bits: 200

# measurement types --> angle zero centered = angle measurement that should be near zero most of the time
# meas_types: ['position','velocity','angle_zero_centered','angle','range','other']
meas_types:
  gps: ['position','position','position']
  lin_rel: ['position','position','position']
  depth: ['position']
  compass: ['angle']
  usbl: ['range','angle_zero_centered','angle_zero_centered']

position:
  range: [-200,200]   # nominal operating area
  resolution: 0.0001  # resolution for uniform quantization
  num_bins: 100       # number of allowed bins for adaptive scaling --> determines scale factor for adaptive binning
  center: 0           # center of expected values for adaptive binning

velocity:
  range: [-5,5]       # nominal operating area
  resolution: 0.00001 # resolution for uniform quantization
  num_bins: 100       # number of allowed bins for adaptive scaling --> determines scale factor for adaptive binning
  center: 0           # center of expected values for adaptive binning

angle:
  range: [0,6.28318530718]  # 0-2pi rad, # nominal operating area
  resolution: 0.00001      # resolution for uniform quantization
  num_bins: 100             # number of allowed bins for adaptive scaling --> determines scale factor for adaptive binning
  center: 3.14159265359     # center of expected values for adaptive binning

angle_zero_centered:
  range: [-3.14159265359, 3.14159265359]  # nominal operating area
  resolution: 0.00001                   # resolution for uniform quantization
  num_bins: 100                           # number of allowed bins for adaptive scaling --> determines scale factor for adaptive binning
  center: 0                               # center of expected values for adaptive binning

range:
  range: [0,500]      # nominal operating area
  resolution: 0.0001  # resolution for uniform quantization
  num_bins: 100       # number of allowed bins for adaptive scaling --> determines scale factor for adaptive binning
  center: 10          # center of expected values for adaptive binning